src/App.tsx

```diff
@@ -1,73 +1,133 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/indent */
+import React, { useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import { Person } from './types/Person';
+import { useEffect, useRef } from 'react';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputFocused, setInputFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [debounceDelay, setDebounceDelay] = useState(300); // Customizable delay
```

Comment:
The `debounceDelay` state is defined but not used dynamically. If the delay is not meant to be changed during runtime, consider using a constant instead of a state variable to simplify the code.